### PR TITLE
Add artifact store in configuration

### PR DESCRIPTION
**Pull request by moria97**
## Description
Add artifact store support (currently [OCI artifacts](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-oci-artifacts)):

- User can config artifacts at image level with imageName and digest (previous only registry level)
- Artifact config can be shared by other features like de-id export (store config files in ACR)

## Related issues
Addresses [#AB80205](https://microsofthealth.visualstudio.com/Health/_workitems/edit/80205/).

## Testing
Unit tests.

## FHIR Team Checklist
- [x] **Update the title** of the PR to be succinct and less than 50 characters
- [x] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [x] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [x] Tag the PR with **Azure API for FHIR** if this will release to the managed service
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Skip

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Core/Features/ArtifactStore/OciArtifactInfo.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+using System;
+
+namespace Microsoft.Health.Fhir.Core.Features.ArtifactStore
+{
+    public class OciArtifactInfo
```
Comment (Ivanidzo4ka) [inline src/Microsoft.Health.Fhir.Core/Features/ArtifactStore/OciArtifactInfo.cs:9]: >Oci [](http://example.com/codeflow?start=17&length=3)

Oracle cloud infrastructure?

This is what I'm getting if I search 'oci azure'

File: src/Microsoft.Health.Fhir.Api/Configs/FhirServerConfiguration.cs [hunk 2]
```diff
@@ -29,5 +29,7 @@ public class FhirServerConfiguration : IApiConfiguration
         public BundleConfiguration Bundle { get; } = new BundleConfiguration();
 
         public ThrottlingConfiguration Throttling { get; } = new ThrottlingConfiguration();
+
+        public ArtifactStoreConfiguration ArtifactStore { get; } = new ArtifactStoreConfiguration();
```
Comment (Ivanidzo4ka) [inline src/Microsoft.Health.Fhir.Api/Configs/FhirServerConfiguration.cs:33]: >ArtifactStore [](http://example.com/codeflow?start=42&length=13)

If it's sole purpose to serve Convert operation, shouldn't it be part of Convert configuration?

File: src/Microsoft.Health.Fhir.Core/Features/ArtifactStore/OciArtifactInfo.cs (hunk 1 above)
Comment (Ivanidzo4ka) [reply src/Microsoft.Health.Fhir.Core/Features/ArtifactStore/OciArtifactInfo.cs:9]: Should read PR description.
Any chance you can add [https://docs.microsoft.com/en-us/azure/container-registry/container-registry-oci-artifacts](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-oci-artifacts) as comment in this file?

Comment (moria97) [reply src/Microsoft.Health.Fhir.Core/Features/ArtifactStore/OciArtifactInfo.cs:11]: Sure. I have added the OCI description to comments.

File: src/Microsoft.Health.Fhir.Api/Configs/FhirServerConfiguration.cs (hunk 2 above)
Comment (moria97) [reply src/Microsoft.Health.Fhir.Api/Configs/FhirServerConfiguration.cs:33]: The previous config inside ConvertData is to serve convert operation only , but this service-level artifact registration will be used by convert data and anonymized export ([#AB81761(https://microsofthealth.visualstudio.com/Health/_backlogs/backlog/Compass/Stories/?workitem=81761)], and more scenarios requires external resources in the future

**Review by Ivanidzo4ka — APPROVED** [review]
:shipit:

**Pull request merged**